Packs/LogPoint_SIEM_Integration/Integrations/LogPoint_SIEM_Integration/LogPoint_SIEM_Integration.py

```diff
@@ -729,10 +733,14 @@ def get_searchid_command(client, args):
 
 def search_logs_command(client, args):
     search_id = args.get('search_id')
-    search_result = client.get_search_results(search_id)
-    if not search_result.get('success'):
-        raise DemistoException(search_result.get('message'))
-    rows = search_result.get('rows', [])
+    rows = []
+    while True:
+        search_result = client.get_search_results(search_id)
+        if not search_result.get('success'):
+            raise DemistoException(search_result.get('message'))
+        rows += search_result.get('rows', [])
+        if search_result.get('final'):
+            break
```

Comment:
**while True** can be harmful if we don't force exit the loop, are we certain that it will always exit? Maybe we can add a counter, or even time it, which will force the while loop to exit

Reply:
You can be sure that there will be an exit, the value "final" is always returned, see the logpoint documentation opposite: https://docs.logpoint.com/docs/logpoint-api-reference/en/latest/Search%20API/getsearchlogs.html